web/src/components/organisms/Settings/Workspace/hooks.ts

```diff
@@ -0,0 +1,225 @@
+import {
+  useGetWorkspacesQuery,
+  useCreateWorkspaceMutation,
+  useUpdateWorkspaceMutation,
+  useDeleteWorkspaceMutation,
+  useAddMemberToWorkspaceMutation,
+  useUpdateMemberOfWorkspaceMutation,
+  Role,
+  useRemoveMemberFromWorkspaceMutation,
+  Workspace,
+  useGetUserBySearchLazyQuery,
+} from "@reearth-cms/gql/graphql-client-api";
+import { useWorkspace } from "@reearth-cms/state";
+import { useCallback, useEffect, useState } from "react";
+import { useNavigate } from "react-router-dom";
+
+export type RoleUnion = "READER" | "WRITER" | "OWNER";
+
+type Params = {
+  workspaceId: string | undefined;
+};
+
+export default (params: Params) => {
```

Comment:
Can you destructure your props and for consistency can you rename the type to `Props`?